distributed-tracing/node-js-app/beachshirt/controllers/deliveryController.js

```diff
@@ -0,0 +1,72 @@
+/**
+* Driver for delivery service which is responsible to deliver the shirts of a given style.
+* @author Yogesh Prasad Kurmi (ykurmi@vmware.com)
+*/
+const uuid4 = require('uuid4');
+const { Tags, FORMAT_HTTP_HEADERS } = require('opentracing');
+
+module.exports = (app, log, tracer) => {
+
+    app.post('/dispatch/:orderNum/', (req, res) => {
+        const parentSpanContext = tracer.extract(FORMAT_HTTP_HEADERS, req.headers)
+        let span = tracer.startSpan('/dispatch', {
+            childOf: parentSpanContext,
+            tags: {[Tags.SPAN_KIND]: Tags.SPAN_KIND_RPC_SERVER}
+        });
+        let orderNum = req.params.orderNum;
+        let shirts = JSON.parse(req.body.shirts);
+        if (Math.floor(Math.random() * 5) === 0) {
+            const error = "Failed to dispatch shirts!";
+            log.error(error);
+            span.setTag(Tags.HTTP_STATUS_CODE, 503)
+            span.setTag(Tags.ERROR, true)
+            span.finish();
+            return res.status(503).json({ error: error });
+        }
+
+        if (Math.floor(Math.random() * 10) === 0) {
+            const error = "Invalid Order Num";
+            log.error(error);
+            span.setTag(Tags.HTTP_STATUS_CODE, 400)
+            span.setTag(Tags.ERROR, true)
+            span.finish();
+            return res.status(400).json({ error: error });
+        }
+
+        if (Math.floor(Math.random() * 10) === 0) {
+            shirts = null;
+        }
+        if (shirts === null || shirts.length === 0){
+            const error = "No shirts to deliver";
+            log.info(error);
+            span.setTag(Tags.HTTP_STATUS_CODE, 400)
+            span.setTag(Tags.ERROR, true)
+            span.finish();
+            return res.status(400).json({ error: error });
+        }
+        trackingNum = uuid4()
+
+        log.debug(`Tracking number of Order:${orderNum} is ${trackingNum}`);
+        setTimeout(() => {
+            span.setTag(Tags.HTTP_STATUS_CODE, 200)
+            span.finish();
+            return res.json({orderNum, trackingNum, message:"shirts delivery dispatched"})
+        }, 1000);
+    });
+
+    app.post('/retrieve/:orderNum/', (req, res) => {
+        let span = tracer.startSpan('/retrieve');
+        let orderNum = req.params.orderNum;
+        if (orderNum === null){
+            const error = "Invalid Order Num";
+            log.error(error);
+            span.setTag(Tags.HTTP_STATUS_CODE, 400)
+            span.setTag(Tags.ERROR, true)
+            span.finish();
+            return res.status(400).json({ error: error });
+        }
+        span.setTag(Tags.HTTP_STATUS_CODE, 200)
+        span.finish();
+        return res.json({message:`Order: ${orderNum} returned`})
+    });
+};
```

Comment:
1. Minor but you can define the following variables as const instead of let since you are not reassigning new values to them:
span on L12 and L58
orderNum on L16 and L59

It's minor but it let's the code reader know that you don't intend to reassign these variables.

2. Everywhere you have return res.status(xxx).json({ **error: error** });  can be rewritten as return res.status(xxx).json({ **error** });
    It's not a big deal, it's just less typing and less to read.